transaction-scenarios/src/scenarios/fungible_resource.rs

```diff
@@ -0,0 +1,435 @@
+                },
+                |core, config, state, result| {
+                    Ok(())
+                },
+            )
+            .failed_transaction_with_error_handler(
+                |core, config, state| {
+                    core.next_transaction_with_faucet_lock_fee(
+                        "fungible-min-div-create-proof-wrong-granularity",
+                        |builder| {
+                            builder.create_proof_from_account_of_amount(
+                                config.user_account_1.address,
+                                state.min_divisibility_fungible_resource.unwrap(),
+                                dec!("0.0001"),
+                            )
+                        },
+                        vec![&config.user_account_1.key],
+                    )
+                },
+                |core, config, state, error| {
+                    Ok(())
+                },
+            )
+            .successful_transaction_with_result_handler(
+                |core, config, state| {
+                    core.next_transaction_with_faucet_lock_fee(
+                        "fungible-min-div-recall-correct-granularity",
+                        |builder| {
+                            builder
+                                .recall(state.vault2.unwrap(), dec!("2"))
+                                .try_deposit_batch_or_abort(config.user_account_1.address)
+                        },
+                        vec![&config.user_account_1.key],
+                    )
+                },
+                |core, config, state, result| {
+                    Ok(())
+                },
+            )
+            .failed_transaction_with_error_handler(
+                |core, config, state| {
+                    core.next_transaction_with_faucet_lock_fee(
+                        "fungible-min-div-recall-wrong-granularity",
+                        |builder| {
+                            builder
+                                .recall(state.vault2.unwrap(), dec!("123.12321"))
+                                .try_deposit_batch_or_abort(config.user_account_1.address)
+                        },
+                        vec![&config.user_account_1.key],
+                    )
+                },
+                |core, config, state, error| {
+                    Ok(())
+                },
+            )
+            .finalize(|core, config, state| {
+                Ok(ScenarioOutput {
+                    interesting_addresses: DescribedAddresses::new()
+                    .add("user_account_1", config.user_account_1.address.clone())
+                    .add("user_account_2", config.user_account_2.address.clone())
+                    .add(
+                        "max_divisibility_fungible_resource",
+                        state.max_divisibility_fungible_resource.unwrap(),
+                    )
+                    .add(
+                        "min_divisibility_fungible_resource",
```

Comment:
Would be quite cool if we leave a vault frozen and we include the vault address here - for the dashboard team -- I guess we need some kind of `AnyAddress` model for that though.

Reply:
TBH we could probably just have these Addresses String instead of an Address object, which would allow us to include EG vault addresses and eg NonFungibleGlobalIds :) - or maybe we could create an Enum:
```
pub enum DescribedAddress {
    Global(GlobalAddress),
    Internal(InternalAddress),
    NonFungible(NonFungibleGlobalId),
}
```